include/lo2s/types.hpp

```diff
@@ -256,6 +257,96 @@ class Package
 private:
     int id_;
 };
+
+class WeakFd;
+class Fd
+{
+public:
+    explicit Fd(int fd) : fd_(fd)
+    {
+    }
+
```

Comment:
A static method `open()` with the signature of the linux open syscall and `std::optional<Fd>` would be nice.